src/Stratis.Bitcoin/Consensus/ConsensusManger.cs

```diff
@@ -83,43 +85,44 @@ public class ConsensusManager
             this.reorgLock = new object();
             this.blockRequestedLock = new object();
 
-            this.blocksRequested = new Dictionary<uint256, List<OnBlockDownloadedCallback>>();
+            this.callbacksByBlocksRequesteHash = new Dictionary<uint256, List<OnBlockDownloadedCallback>>();
             this.toDownloadQueue = new Queue<BlockDownloadRequest>();
         }
 
         /// <summary>
-        /// Set the tip of <see cref="ConsensusManager"/> to be the header that is stored to disk,
-        /// if the given <see cref="consensusTip"/> is not in store then rewind store till a tip is found.
-        /// If <see cref="IChainState.BlockStoreTip"/> is not null (block store is enabled) ensure the store and consensus manager are on the same tip,
-        /// if consensus manager is ahead then reorg it to be the same height as store.
+        /// Set the tip of <see cref="ConsensusManager"/>, if the given <paramref name="chainTip"/> is not similar to <see cref="Tip"/>
+        /// then rewind consensus until a common header is found.
         /// </summary>
-        /// <param name="consensusTip">The consensus tip that is attempted to be set.</param>
-        public async Task InitializeAsync(ChainedHeader consensusTip)
+        /// <remarks>
+        /// If <see cref="blockStore"/> is not null (block store is enabled) then all block headers in <see cref="IChainedHeaderTree"/> will be marked as blocks are available.
```

Comment:
`<c>`

Reply:
=> `marked as their block data is available.`

and maybe not `IChainedHeaderTree` but our instance `chainedHeaderTree`